deployment-tinkerer/src/main/java/org/wso2/testgrid/deployment/tinkerer/providers/AWSProvider.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.deployment.tinkerer.providers;
+
+import com.amazonaws.auth.PropertiesFileCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.AmazonEC2ClientBuilder;
+import com.amazonaws.services.ec2.model.DescribeInstancesRequest;
+import com.amazonaws.services.ec2.model.DescribeInstancesResult;
+import com.amazonaws.services.ec2.model.Tag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.util.TestGridUtil;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Implementation of the AWS cloud provider functionality.
+ *
+ * @since 1.0.0
+ */
+public class AWSProvider implements Provider {
+
+    private static final Logger logger = LoggerFactory.getLogger(AWSProvider.class);
+
+    private static Path configFilePath;
+
+    static {
+        try {
+            configFilePath = TestGridUtil.getConfigFilePath();
+        } catch (IOException e) {
+            logger.error("Error occurred while getting the config path.", e);
+        }
+    }
+
+    @Override
+    public String getInstanceName(String region, String instanceId) {
```

Comment:
*getEC2InstanceName.

Reply:
This method should be generic as it defined in the Provider interface and intended to be generic across multiple providers like Azure, GCE